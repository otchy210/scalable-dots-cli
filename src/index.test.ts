import { sayHello } from '.';

describe('sayHello', () => {
  it('works', () => {
    const log = jest.spyOn(console, 'log');
    sayHello();
    expect(log).toBeCalledWith('Hello Otchy!');
  });
});
