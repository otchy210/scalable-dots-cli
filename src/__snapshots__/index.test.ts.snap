// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToXml matches snapshot when options are default 1`] = `"<svg viewBox=\\"0 0 273 273\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"><path d=\\"m0 0h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m17 0h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m68 0h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m85 0h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m136 0h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m153 0h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m238 0h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m255 0h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m0 17h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m17 17h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m68 17h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m85 17h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m136 17h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m153 17h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m238 17h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m255 17h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m34 34h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m51 34h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m102 34h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m119 34h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m204 34h16v16h-16z\\" fill=\\"#00ffff\\"/><path d=\\"m221 34h16v16h-16z\\" fill=\\"#00ffff\\"/><path d=\\"m34 51h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m51 51h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m102 51h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m119 51h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m204 51h16v16h-16z\\" fill=\\"#00ffff\\"/><path d=\\"m221 51h16v16h-16z\\" fill=\\"#00ffff\\"/><path d=\\"m0 68h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m17 68h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m68 68h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m85 68h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m170 68h16v16h-16z\\" fill=\\"#0000ff\\"/><path d=\\"m187 68h16v16h-16z\\" fill=\\"#0000ff\\"/><path d=\\"m0 85h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m17 85h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m68 85h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m85 85h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m170 85h16v16h-16z\\" fill=\\"#0000ff\\"/><path d=\\"m187 85h16v16h-16z\\" fill=\\"#0000ff\\"/><path d=\\"m34 102h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m51 102h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m102 102h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m119 102h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m136 102h16v16h-16z\\" fill=\\"#ff0000\\"/><path d=\\"m153 102h16v16h-16z\\" fill=\\"#ff0000\\"/><path d=\\"m238 102h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m255 102h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m34 119h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m51 119h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m102 119h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m119 119h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m136 119h16v16h-16z\\" fill=\\"#ff0000\\"/><path d=\\"m153 119h16v16h-16z\\" fill=\\"#ff0000\\"/><path d=\\"m238 119h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m255 119h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/><path d=\\"m0 136h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m17 136h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m102 136h16v16h-16z\\" fill=\\"#00ff00\\"/><path d=\\"m119 136h16v16h-16z\\" fill=\\"#00ff00\\"/><path d=\\"m136 136h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m153 136h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m204 136h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m221 136h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m0 153h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m17 153h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m102 153h16v16h-16z\\" fill=\\"#00ff00\\"/><path d=\\"m119 153h16v16h-16z\\" fill=\\"#00ff00\\"/><path d=\\"m136 153h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m153 153h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m204 153h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m221 153h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/><path d=\\"m68 170h16v16h-16z\\" fill=\\"#ffff00\\"/><path d=\\"m85 170h16v16h-16z\\" fill=\\"#ffff00\\"/><path d=\\"m170 170h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m187 170h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m238 170h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m255 170h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m68 187h16v16h-16z\\" fill=\\"#ffff00\\"/><path d=\\"m85 187h16v16h-16z\\" fill=\\"#ffff00\\"/><path d=\\"m170 187h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m187 187h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m238 187h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m255 187h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/><path d=\\"m34 204h16v16h-16z\\" fill=\\"#ff00ff\\"/><path d=\\"m51 204h16v16h-16z\\" fill=\\"#ff00ff\\"/><path d=\\"m136 204h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m153 204h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m204 204h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m221 204h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m34 221h16v16h-16z\\" fill=\\"#ff00ff\\"/><path d=\\"m51 221h16v16h-16z\\" fill=\\"#ff00ff\\"/><path d=\\"m136 221h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m153 221h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/><path d=\\"m204 221h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m221 221h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m0 238h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m17 238h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m102 238h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m119 238h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m170 238h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m187 238h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m238 238h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m255 238h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m0 255h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m17 255h16v16h-16z\\" fill=\\"#ffffff\\"/><path d=\\"m102 255h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m119 255h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/><path d=\\"m170 255h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m187 255h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/><path d=\\"m238 255h16v16h-16z\\" fill=\\"#000000\\"/><path d=\\"m255 255h16v16h-16z\\" fill=\\"#000000\\"/></svg>"`;

exports[`convertToXml matches snapshot when options enable prettyPrint 1`] = `
"<svg viewBox=\\"0 0 273 273\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\">
  <path d=\\"m0 0h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m17 0h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m68 0h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m85 0h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m136 0h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m153 0h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m238 0h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m255 0h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m0 17h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m17 17h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m68 17h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m85 17h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m136 17h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m153 17h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m238 17h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m255 17h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m34 34h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m51 34h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m102 34h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m119 34h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m204 34h16v16h-16z\\" fill=\\"#00ffff\\"/>
  <path d=\\"m221 34h16v16h-16z\\" fill=\\"#00ffff\\"/>
  <path d=\\"m34 51h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m51 51h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m102 51h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m119 51h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m204 51h16v16h-16z\\" fill=\\"#00ffff\\"/>
  <path d=\\"m221 51h16v16h-16z\\" fill=\\"#00ffff\\"/>
  <path d=\\"m0 68h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m17 68h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m68 68h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m85 68h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m170 68h16v16h-16z\\" fill=\\"#0000ff\\"/>
  <path d=\\"m187 68h16v16h-16z\\" fill=\\"#0000ff\\"/>
  <path d=\\"m0 85h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m17 85h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m68 85h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m85 85h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m170 85h16v16h-16z\\" fill=\\"#0000ff\\"/>
  <path d=\\"m187 85h16v16h-16z\\" fill=\\"#0000ff\\"/>
  <path d=\\"m34 102h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m51 102h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m102 102h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m119 102h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m136 102h16v16h-16z\\" fill=\\"#ff0000\\"/>
  <path d=\\"m153 102h16v16h-16z\\" fill=\\"#ff0000\\"/>
  <path d=\\"m238 102h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m255 102h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m34 119h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m51 119h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m102 119h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m119 119h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m136 119h16v16h-16z\\" fill=\\"#ff0000\\"/>
  <path d=\\"m153 119h16v16h-16z\\" fill=\\"#ff0000\\"/>
  <path d=\\"m238 119h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m255 119h16v16h-16z\\" fill=\\"rgba(0,0,255,.496)\\"/>
  <path d=\\"m0 136h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m17 136h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m102 136h16v16h-16z\\" fill=\\"#00ff00\\"/>
  <path d=\\"m119 136h16v16h-16z\\" fill=\\"#00ff00\\"/>
  <path d=\\"m136 136h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m153 136h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m204 136h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m221 136h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m0 153h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m17 153h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m102 153h16v16h-16z\\" fill=\\"#00ff00\\"/>
  <path d=\\"m119 153h16v16h-16z\\" fill=\\"#00ff00\\"/>
  <path d=\\"m136 153h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m153 153h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m204 153h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m221 153h16v16h-16z\\" fill=\\"rgba(255,0,0,.496)\\"/>
  <path d=\\"m68 170h16v16h-16z\\" fill=\\"#ffff00\\"/>
  <path d=\\"m85 170h16v16h-16z\\" fill=\\"#ffff00\\"/>
  <path d=\\"m170 170h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m187 170h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m238 170h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m255 170h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m68 187h16v16h-16z\\" fill=\\"#ffff00\\"/>
  <path d=\\"m85 187h16v16h-16z\\" fill=\\"#ffff00\\"/>
  <path d=\\"m170 187h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m187 187h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m238 187h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m255 187h16v16h-16z\\" fill=\\"rgba(0,255,255,.199)\\"/>
  <path d=\\"m34 204h16v16h-16z\\" fill=\\"#ff00ff\\"/>
  <path d=\\"m51 204h16v16h-16z\\" fill=\\"#ff00ff\\"/>
  <path d=\\"m136 204h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m153 204h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m204 204h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m221 204h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m34 221h16v16h-16z\\" fill=\\"#ff00ff\\"/>
  <path d=\\"m51 221h16v16h-16z\\" fill=\\"#ff00ff\\"/>
  <path d=\\"m136 221h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m153 221h16v16h-16z\\" fill=\\"rgba(0,255,0,.496)\\"/>
  <path d=\\"m204 221h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m221 221h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m0 238h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m17 238h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m102 238h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m119 238h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m170 238h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m187 238h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m238 238h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m255 238h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m0 255h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m17 255h16v16h-16z\\" fill=\\"#ffffff\\"/>
  <path d=\\"m102 255h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m119 255h16v16h-16z\\" fill=\\"rgba(255,255,0,.496)\\"/>
  <path d=\\"m170 255h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m187 255h16v16h-16z\\" fill=\\"rgba(255,0,255,.199)\\"/>
  <path d=\\"m238 255h16v16h-16z\\" fill=\\"#000000\\"/>
  <path d=\\"m255 255h16v16h-16z\\" fill=\\"#000000\\"/>
</svg>"
`;
